crypto/bio/bio_meth.c

```diff
@@ -9,13 +9,14 @@
 
 #include "bio_lcl.h"
 #include "internal/thread_once.h"
+#include "internal/glock.h"
 
 CRYPTO_RWLOCK *bio_type_lock = NULL;
 static CRYPTO_ONCE bio_type_init = CRYPTO_ONCE_STATIC_INIT;
 
 DEFINE_RUN_ONCE_STATIC(do_bio_type_init)
 {
-    bio_type_lock = CRYPTO_THREAD_glock_new("bio_type");
+    bio_type_lock = global_locks[CRYPTO_GLOCK_BIO_TYPE];
```

Comment:
It still might be appropriate to make it a function call. I mean with enum as argument, as alternative to indexing an array.